### PR TITLE
[CDAP-14116] Verify that all licenses are included in all distributions

### PR DESCRIPTION
**Pull request by mattwuenschel**
This PR adds the generic licenses to distributed and sandbox packages of CDAP. This will also add the third party specific artifacts/licenses for licenses that require source to be included. Each license was pulled from their respective Maven repo. 

This will package the jars from the local Maven repository (${user.home}/.m2/repository). https://maven.apache.org/settings.html. All versions of an artifact pulled from Maven will be packaged with CDAP and any future version updates will automatically be included. With the current artifacts, this will add 9mb to each distribution. 

Artifacts and libraries that require source to be included:
- javax.activation.activation-1.1.jar
- javax.annotation.jsr250-api-1.0.jar
- javax.servlet.jsp.jsp-api
- javax.servlet.javax.servlet-api-3.0.1.jar
- javax.ws.rs.javax.ws.rs-api-2.0.jar
- javax.xml.bind.jaxb-api
- jaxb-impl
- net.jcip.jcip-annotations-1.0.jar
- jersey
- com.sun.jersey.jersey-client-1.9.jar (merged into jersey folder)
- com.sun.jersey.jersey-core-1.9.jar (merged into jersey folder)
- com.sun.jersey.jersey-server-1.9.jar (merged into jersey folder)
- javax.xml.stream.stax-api

Example output
```
$ ls
artifacts  bin  bootstrap  conf  examples  ext  lib  libexec  LICENSES  README  third_party  ui  VERSION
$ ls LICENSES/
ANTLR  APACHE  Apache-LICENSE-2.0  BSD  CCO  CDDL  EPL  MIT  W3C
$ ls third_party/
activation  annotation  javax-servlet-api  javax-servlet-jsp-api  javax-ws-rs-api  jaxb-api  jaxb-impl  jcip-annotations  jersey  jta  stax-api
$ ls third_party/activation/
1.1  LICENSE
$ ls third_party/activation/1.1/
activation-1.1.jar
```

Packaging updates:
- [x] Add generic license to LICENSE folder
- [x] Add artifact specific license to third_party folder
- [x] Add source artifacts to third_party folder

Test run:
- [x]  Test SDK build
- [x]  Test distributed package builds (rpm, deb)
- [x]  Test parcel build

JIRAs:
Add license dependencies to SDK
https://issues.cask.co/browse/CDAP-14082
Verify that all licenses are included in the all distributions
https://issues.cask.co/browse/CDAP-14116

### CODE REVIEW

**Comment by mattwuenschel** [comment]
One more note, the packaging changes are in cdap-standalone/pom.xml and pom.xml. Everything else is importing licenses.

File: cdap-standalone/pom.xml [hunk 1]
```diff
@@ -531,6 +531,96 @@
...
+                    <!--copy third party artifacts-->
+                    <resource>
+                      <directory>${project.basedir}/src/dist/third_party</directory>
+                      <targetPath>third_party</targetPath>
+                    </resource>
+                    <resource>
+                      <directory>${user.home}/.m2/repository/javax/activation/activation</directory>
+                      <targetPath>third_party/activation</targetPath>
+                      <includes>
+                        <include>**/*.jar</include>
```
Comment (sreevatsanraman) [inline cdap-standalone/pom.xml:543]: Are we including all the jars? Shouldn't this include only source jars?

File: cdap-standalone/pom.xml [hunk 2]
```diff
@@ -531,6 +531,96 @@
                       <directory>${project.basedir}/src/dist/LICENSES</directory>
                       <targetPath>LICENSES</targetPath>
                     </resource>
+                    <!--copy third party artifacts-->
+                    <resource>
+                      <directory>${project.basedir}/src/dist/third_party</directory>
+                      <targetPath>third_party</targetPath>
+                    </resource>
+                    <resource>
+                      <directory>${user.home}/.m2/repository/javax/activation/activation</directory>
```
Comment (chtyim) [inline cdap-standalone/pom.xml:540]: Don't do this. This will break if the maven repo is set to something else.

File: pom.xml [hunk 3]
```diff
@@ -2140,6 +2140,100 @@
                           <include>LICENSE-2.0.txt</include>
                         </includes>
                       </resource>
+                      <resource>
```
Comment (chtyim) [inline pom.xml:2143]: Why these are repeated in cdap-standalone/pom.xml??

**Comment by mattwuenschel** [comment]
Closing as an edge case was brought up that this wouldn't work for. While `${user.home}/.m2/repository` is documented on mavens site, this can be changed and would break one of the builds.

**Pull request closed without merging**